### PR TITLE
Add storage V6 migration to v0.23

### PR DESCRIPTION
**Pull request by SupunS**
Adding back the storage migration V6 to `v0.23` that was left out when merging to master.

This simply copies the two files (`storage_v6.go` and `storage_v6_test.go`) from the [v0.23-testnet](https://github.com/onflow/flow-go/tree/v0.23-testnet) branch.

### CODE REVIEW

File: cmd/util/cmd/execution-state-extract/execution_state_extract.go [hunk 1]
```diff
@@ -66,11 +66,17 @@ func extractExecutionState(
...
+			Log:       log,
+			OutputDir: outputDir,
+		}
+
 		migrations = []ledger.Migration{
-			storageUsedUpdateMigration.Migrate,
 			mgr.PruneMigration,
+			storageFormatV6Migration.Migrate,
+			storageUsedUpdateMigration.Migrate,
+			mgr.PruneMigration, // we need post migration pruning before deploying trie update pruning
```
Comment (SupunS) [inline cmd/util/cmd/execution-state-extract/execution_state_extract.go:78]: I added this post-migration pruning step, similar to what we had in `v0.22-testnet`:
https://github.com/onflow/flow-go/blob/eaf0e5c1e20cde49b94a7ed6bc1768af19bfd3a8/cmd/util/cmd/execution-state-extract/execution_state_extract.go#L72-L76

**Review by turbolent — APPROVED** [review]
🎉

File: cmd/util/ledger/reporters/account_reporter.go [hunk 2]
```diff
@@ -64,7 +65,10 @@ func (r *AccountReporter) Report(payload []ledger.Payload) error {
 	defer rwm.Close()
 
 	l := migrations.NewView(payload)
-	st := state.NewState(l)
+	st := state.NewState(
+		l,
+		state.WithMaxInteractionSizeAllowed(math.MaxUint64),
+	)
```
Comment (SupunS) [inline cmd/util/ledger/reporters/account_reporter.go:71]: Had to increase the `MaxInteractionSizeAllowed` here and below.
cc: @janezpodhostnik

File: cmd/util/ledger/migrations/storage_v6.go [hunk 3]
```diff
@@ -0,0 +1,3773 @@
...
+	deferredValuePaths := make(map[storagePath]bool)
+	for _, payload := range payloads {
+		m.incrementProgress()
+
+		keyParts := payload.Key.KeyParts
+		rawOwner := keyParts[0].Value
+		rawController := keyParts[1].Value
+		rawKey := keyParts[2].Value
+
+		if state.IsFVMStateKey(
```
Comment (janezpodhostnik) [inline cmd/util/ledger/migrations/storage_v6.go:276]: before calling this function the payloads were split: `fvmPayloads, storagePayloads, slabPayloads := splitPayloads(payloads)`

would it make sense here to only use ` storagePayloads` and `slabPayloads` here and skip this if condition?

**Review by janezpodhostnik — APPROVED** [review]
Overall looks good to me. I did not review the smart contracts.

File: cmd/util/ledger/migrations/storage_v6.go (hunk 3 above)
Comment (SupunS) [reply cmd/util/ledger/migrations/storage_v6.go:283]: yeah, good idea!
We didn't have the split function earlier, and this wasn't cleaned up once it was introduced. good catch!

**Review by turbolent — APPROVED** [review]
👍

**Pull request merged**